compiler/src/iree/compiler/Codegen/Common/FoldTensorSubsetIntoVectorTransferOps.cpp

```diff
@@ -68,6 +68,26 @@ class FoldExtractSliceIntoTransferRead final
       return failure();
     if (!extractOp.hasUnitStride())
       return failure();
+    // When there's more than 1 use of the result of tensor.extract_slice,
+    // folding extractOp + xferOp won't eliminate the need for the former. So
+    // this transformation wouldn't be beneficial. It would, however, generate
+    // another "user" of the original tensor, which then would prevent the
+    // hoisting logic to hoist the following pair in the example below:
+    //  * "tensor.extract_slice" + "tensor.insert_slice"
+    // ```mlir
+    // %res = scf.for {{.*}} iter_args(%loop_arg = %extracted_slice) -> {{.*}} {
+    //     %extracted_slice = tensor.extract_slice %loop_arg
+    //     %read = vector.transfer_read %extracted_slice
+    //     // Some opeeration using %read that generates %res
+    //     %write = vector.transfer_write %res, %extracted_slice
+    //     %inserted_slice = tensor.insert_slice %write into %loop_arg
+    //     scf.yield %inserted_slice
+    //  }
+    //```
+    for (auto user : extractOp.getResult().getUsers()) {
```

Comment:
nit: can we use the exact type instead of auto? It's not immediately clear what the type is and the return type being a typedef like `iterator_range<user_iterator>` doesn't help much. I think it should be:
```suggestion
    for (Operation *user : extractOp.getResult().getUsers()) {
```

Reply:
nit2: I think we could also do:
```c++
if (!extractOp.getResult().hasOneUse())
  return failure():
```
since the extract op is one of the operands of `xfer` and I don't think it could be used twice by the same op?

Reply:
👍🏻 Yup, that felt like a big hammer 😅